utils/sysfs/sysfs.go

```diff
@@ -335,25 +348,198 @@ func (fs *realSysFs) GetSystemUUID() (string, error) {
+	}
+	return realState == desiredState
+}
+
+// getOnlineStateID looks for line similar to `206: online` from /sys/devices/system/cpu/hotplug/states
+// and extracts number from the beginning.
+// See: https://www.kernel.org/doc/html/latest/core-api/cpu_hotplug.html#testing-of-hotplug-states
+func getOnlineStateID() (int, error) {
+	const (
+		lf          byte = 0x0A
+		colon       byte = 0x3A
+		onlineState      = "online"
+	)
+
+	states, err := ioutil.ReadFile(hotplugStates)
+	if err != nil {
+		return 0, err
+	}
+	var online int
+	statesByLine := bytes.Split(states, []byte{lf})
+
+	// We start from last row as it's usually where online can be found.
+	for i := len(statesByLine) - 1; i >= 0; i-- {
+		// Line in the file looks like: `206: online`. We need to split first.
+		state := bytes.Split(statesByLine[i], []byte{colon})
+		// Exactly two elements are expected.
+		if len(state) != 2 {
+			continue
+		}
+		if string(bytes.TrimSpace(state[1])) == onlineState {
+			// Value of trimmed second element matches `online`.
+			online, err = strconv.Atoi(string(state[0]))
+			if err != nil {
+				return 0, err
+			}
+			break
+		}
+	}
+	return online, nil
+}
+
+// Looks for sysfs cpu path containing given CPU property, e.g. core_id or physical_package_id
+// and returns number of unique values of given property, exemplary usage: getting number of CPU physical cores
+func GetUniqueCPUPropertyCount(cpuBusPath string, propertyName string) int {
+	pathPattern := cpuBusPath + "cpu*[0-9]"
+	sysCPUPaths, err := filepath.Glob(pathPattern)
+	if err != nil {
+		klog.Errorf("Cannot find files matching pattern (pathPattern: %s),  number of unique %s set to 0", pathPattern, propertyName)
+		return 0
+	}
+	uniques := make(map[string]bool)
+	for _, sysCPUPath := range sysCPUPaths {
+		if !isCPUOnline(sysCPUPath) {
+			continue
+		}
+		propertyPath := filepath.Join(sysCPUPath, sysFsCPUTopology, propertyName)
+		propertyVal, err := ioutil.ReadFile(propertyPath)
+		if err != nil {
+			klog.Errorf("Cannot open %s, number of unique %s  set to 0", propertyPath, propertyName)
+			return 0
+		}
+		packagePath := filepath.Join(sysCPUPath, sysFsCPUTopology, CPUPhysicalPackageID)
+		packageVal, err := ioutil.ReadFile(packagePath)
+		if err != nil {
+			klog.Errorf("Cannot open %s, number of unique %s  set to 0", packagePath, CPUPhysicalPackageID)
+			return 0
+		}
+		uniques[fmt.Sprintf("%s_%s", propertyVal, packageVal)] = true
+	}
+	return len(uniques)
+}
+
+// GetMachineMemoryByType returns information about memory capacity and number of DIMMs.
+// Information is retrieved from sysfs edac per-DIMM API (/sys/devices/system/edac/mc/)
+// introduced in kernel 3.6. Documentation can be found at
+// https://www.kernel.org/doc/Documentation/admin-guide/ras.rst.
+// Full list of memory types can be found in edac_mc.c
+// (https://github.com/torvalds/linux/blob/v5.5/drivers/edac/edac_mc.c#L198)
+func GetMachineMemoryByType(edacPath string) (map[string]*v1.MemoryInfo, error) {
```

Comment:
I know you mentioned you moved `GetMachineMemoryByType` since it makes more sense in `utils/sysfs`, but I would prefer if we can do that in a followup separate PR since this PR is already quite large and focused specifically on CPU topology / online info